src/App.tsx

```diff
@@ -2,24 +1,66 @@
 import './App.css';
-import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter, Route, Routes, useNavigate } from 'react-router-dom';
+
+import { getNumbers, maybeParseInt } from './utils';
+
+import { useUrlQuery } from './hooks/useUrlQuery';
+
+import { Pagination } from './components/Pagination/Pagination';
+
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
-export const App: React.FC = () => {
+const RoutedApp: React.FC = () => {
+  const query = useUrlQuery();
+  const navigate = useNavigate();
+
+  const [perPage, setPerPage] = useState(
+    maybeParseInt(query.get('perPage')) ?? 5,
+  );
+  const [currentPage, setCurrentPage] = useState(
+    maybeParseInt(query.get('page')) ?? 1,
+  );
+
+  useEffect(() => {
+    if (!query.get('page') || !query.get('perPage')) {
+      navigate(`?page=${currentPage}&perPage=${perPage}`);
+    }
+  }, [perPage, query, navigate, currentPage]);
+
+  const handlePageChange = (newPage: number) => {
+    setCurrentPage(newPage);
+    navigate(`?page=${newPage}&perPage=${perPage}`);
+  };
+
+  const handlePerPageChange = (newPerPage: number) => {
+    setPerPage(newPerPage);
+    setCurrentPage(1);
+
+    navigate(`?page=${currentPage}&perPage=${newPerPage}`);
```

Comment:
When changing the 'perPage' value, the URL is updated with the current page before it's set back to 1 on the next line. This can lead to an inconsistent state where the URL indicates a different page than what is actually being displayed. The URL should be updated after the current page is set to 1.